libmamba/src/core/curl.cpp

```diff
@@ -307,4 +308,180 @@ namespace mamba
         return m_errorbuffer;
     }
 
+    CURL* unwrap(const CURLHandle& h)
+    {
+        return h.m_handle;
+    }
+
+    bool operator==(const CURLHandle& lhs, const CURLHandle& rhs)
+    {
+        return unwrap(lhs) == unwrap(rhs);
+    }
+
+    bool operator!=(const CURLHandle& lhs, const CURLHandle& rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    /*****************
+     * CURLReference *
+     *****************/
+
+    CURLReference::CURLReference(CURL* handle)
+        : p_handle(handle)
+    {
+    }
+
+    CURL* unwrap(const CURLReference& h)
+    {
+        return h.p_handle;
+    }
+
+    bool operator==(const CURLReference& lhs, const CURLReference& rhs)
+    {
+        return unwrap(lhs) == unwrap(rhs);
+    }
+
+    bool operator==(const CURLReference& lhs, const CURLHandle& rhs)
+    {
+        return unwrap(lhs) == unwrap(rhs);
+    }
+
+    bool operator==(const CURLHandle& lhs, const CURLReference& rhs)
+    {
+        return unwrap(lhs) == unwrap(rhs);
+    }
+
+    bool operator!=(const CURLReference& lhs, const CURLReference& rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    bool operator!=(const CURLReference& lhs, const CURLHandle& rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    bool operator!=(const CURLHandle& lhs, const CURLReference& rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    /*******************
+     * CURLMultiHandle *
+     *******************/
+
+    CURLMultiHandle::CURLMultiHandle(std::size_t max_parallel_downloads)
+        : p_handle(curl_multi_init())
+        , m_max_parallel_downloads(max_parallel_downloads)
+    {
+        if (p_handle == nullptr)
+        {
+            throw curl_error("Could not initialize CURL multi handle");
```

Comment:
Can't we use `curl_multi_strerror` to get the error message here?

Reply:
indeed, good catch